aspnetcore/fundamentals/localization-extensibility.md

```diff
@@ -0,0 +1,126 @@
+    if (segments.Length > 1 && segments[0].Length == 2)
+    {
+        currentCulture = segments[0];
+    }
+
+    var requestCulture = new ProviderCultureResult(culture);
+    
+    return Task.FromResult(requestCulture);
+}));
+```
+::: moniker-end
+
+### Using a new implemetation of `RequestCultureProvider`
+As mentioned ealier there are four implementation of `RequestCultureProvider` out of the box, which determines the request culture information from various sources.
+
+The developer can easily use this as extensible point to create a new implementation of `RequestCultureProvider` that determines the request culture information from custom source such as configuration file, database ... etc.
+
+The following example shows `AppSettingsRequestCultureProvider` which extend the `RequestCultureProvider` to determines the request culture information from `appsettings.json`.
+
+```csharp
+public class AppSettingsRequestCultureProvider : RequestCultureProvider
+{
+    public string CultureKey { get; set; } = "culture";
+
+    public string UICultureKey { get; set; } = "ui-culture";
+
+    public override Task<ProviderCultureResult> DetermineProviderCultureResult(HttpContext httpContext)
+    {
+        if (httpContext == null)
+        {
+            throw new ArgumentNullException();
+        }
+
+        var configuration = httpContext.RequestServices.GetService<IConfigurationRoot>();
+        var culture = configuration[CultureKey];
+        var uiCulture = configuration[UICultureKey];
+
+        if (culture == null && uiCulture == null)
+        {
+            return Task.FromResult((ProviderCultureResult)null);
+        }
+
+        if (culture != null && uiCulture == null)
+        {
+            uiCulture = culture;
+        }
+
+        if (culture == null && uiCulture != null)
+        {
+            culture = uiCulture;
+        }
+        
+        var providerResultCulture = new ProviderCultureResult(culture, uiCulture);
+
+        return Task.FromResult(providerResultCulture);
+    }
+}
+```
+
+## Localization Resources
+
+ASP.NET Core localization APIs came up with `ResourceManagerStringLocalizer` which an implementation of `IStringLocalizer` that is using `resx` files as underlying storage for the localization resources.
+
+But you are not limited to use the `resx` files if it isn't fit your need. We Already mentioned ealier that [OrchardCore](https://github.com/orchardCMS/OrchardCore/) introduced `POStringLocalizer` which using `PO` instead, to support pluralization.
+
+We have seen many of the community projects that extends `IStringLocalizer` to create another localizers such as [EFStringLocalizer](https://github.com/aspnet/Entropy/tree/master/samples/Localization.EntityFramework), [JsonStringLocalizer](https://github.com/hishamco/My.Extensions.Localization.Json), [SqlLocalizer](https://github.com/damienbod/AspNetCoreLocalization) .. etc.
```

Comment:
```suggestion
Here are some examples of projects which implement `IStringLocalizer`: [EFStringLocalizer](https://github.com/aspnet/Entropy/tree/master/samples/Localization.EntityFramework), [JsonStringLocalizer](https://github.com/hishamco/My.Extensions.Localization.Json), [SqlLocalizer](https://github.com/damienbod/AspNetCoreLocalization) .. etc.
```

I feel like listing these is a bit fraught. The Entropy sample for example is not actively built anymore, and likely hasn't reacted the API changes we made to `IStringLocalizer` in 3.0. If we do keep that section then I think the suggestion above makes it clear that these are examples of how to implement IStringLocalizer, not just a list of community projects.

Reply:
The second two are the community projects